blockchain/smart_contracts/sc_trusted.py

```diff
@@ -0,0 +1,229 @@
+            # insert new sc into database
+            if self._insert_sc(pl, "tsc", sc_key):
+                status = self._sc_provisioning_helper(pl, "tsc", sc_key)
+        return status
+
+    def provision_ssc(self, transaction):
+        """
+        provision ssc class smart contract
+        :param transaction: transaction to extract sc data from
+        """
+        pl = transaction['payload']
+        criteria = pl['criteria']
+        sc_key = ""
+        if "origin_id" in criteria:
+            if "origin_id" in pl and pl['origin_id']:
+                sc_key += pl['origin_id']
+            else:
+                return False
+        sc_key += ":"
+        if "transaction_type" in criteria:
+            if "transaction_type" in pl and pl['transaction_type']:
+                sc_key += pl['transaction_type']
+            else:
+                return False
+        sc_key += ":"
+        if "phase" in criteria:
+            if "phase" in pl and pl['phase']:
+                sc_key += str(pl['phase'])
+            else:
+                return False
+        # insert new sc into database
+        if not self._insert_sc(pl, "ssc", sc_key):
+            return False
+        return self._sc_provisioning_helper(pl, "ssc", sc_key)
+
+    def provision_lsc(self, transaction):
+        """ provide name and python module to run sc """
+        return True
+
+    def provision_bsc(self, transaction):
+        """
+        provision bsc class smart contract
+        :param transaction: transaction to extract sc data from
+        """
+        status = False
+        pl = transaction['payload']
+        if "phase" in pl and pl['phase']:
+            sc_key = pl['phase']
+            if self._insert_sc(pl, "bsc", sc_key):
+                status = self._sc_provisioning_helper(pl, "bsc", sc_key)
+        return status
+
+    def _sc_provisioning_helper(self, pl, sc_class, sc_key):
+        """
+        insert sc code into appropriate dictionary
+        :param pl: transaction payload to extract sc from
+        :param sc_class: type of sc being dealt with (e.g. tsc, ssc, etc.)
+        """
+        try:
+            if 'smart_contract' in pl:
+                sc = pl['smart_contract']
+                if sc[sc_class]:
+                    func = None
+                    # define sc function
+                    exec(sc[sc_class])
+                    # store sc function for this txn type
+                    self.sc_container[sc_class][sc_key] = func
+                else:
+                    logger().warning("No smart contract code provided...")
+                    return False
+            else:
+                return False
+        except Exception as ex:
+            template = "An exception of type {0} occurred. Arguments:\n{1!r}"
+            message = template.format(type(ex).__name__, ex.args)
+            logger().warning(message)
+            return False
+        return True
+
+    def _insert_sc(self, pl, sc_class, sc_key):
```

Comment:
Interesting idea.
And, yes, we'll need to be able to manually specify that a SC is no longer active, and let the SC be able to do that as well.

We _could_ (should?) make it so that some SC could own other SC and could either completely control others or themselves, or in combination in multi-sig configuration with other SC or humans.